src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -419,12 +419,41 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_predicate_node(
   // TODO(anybody) Complex predicates are not processed right now and statistics objects are forwarded.
   //               That implies estimating a selectivity of 1 for such predicates.
   if (!operator_scan_predicates) {
-    // We might not have resolved the predicate because it contains subqueries. Unfortunately, we do not know the
-    // values of predicates on uncorrelated subquery results before query execution. However, if the predicate has an
-    // equals or between condition, it acts as a filter comparable to a semi-join with the join key of the subquery
-    // result (see examples below). We obtain such predicates with subquery results from the JoinToPredicateRewriteRule.
-    // This rule also checks that all preconditions are met to ensure correct query results. Thus, we do not check them
-    // here. For more information about this query rewrite, see `join_to_predicate_rewrite_rule.hpp`.
+    // We can obtain predicates with subquery results from the JoinToPredicateRewriteRule, which turns (semi-)joins into
+    // predicates. OperatorScanPredicate::from_expression(...) cannot resolve these predicates. They act as a filter
+    // comparable to a semi-join with the join key of the subquery result (see examples below). The optimizer rule
+    // checks that all preconditions are met to ensure correct query results. Especially, it guarantees that the
```

Comment:
Is "it" the rule? Which rule? The current or the previously mentioned JoinToPredicateRewriteRule?